src/Agent.Worker/TestResults/TestRunPublisher.cs

```diff
@@ -289,7 +289,10 @@ private TestAttachmentRequestModel GetAttachmentRequestModel(string attachment)
                 string encodedData = Convert.ToBase64String(bytes);
                 if (encodedData.Length <= TCM_MAX_FILECONTENT_SIZE)
                 {
-                    return new TestAttachmentRequestModel(encodedData, Path.GetFileName(attachment), "", GetAttachmentType(attachment));
+                    string attachmentFileName = Path.GetFileName(attachment);
+                    attachmentFileName = attachmentFileName.Replace(":", "_");
```

Comment:
Please add the comment for why we are doing this only for this character

Reply:
Added